### PR TITLE
[MaterializedView] add create MaterializedView logic [P1]

### LINKED ISSUE #4789: [MaterializedView] add create MaterializedView logic [P1]
## Feature request

**Is your feature request related to a problem? Please describe.**

to #3680

**Describe the solution you'd like**

add logic to create MaterializedView

### PR DESCRIPTION
**Pull request by blackstar-baba**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #4789

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 82 / 132 (62.12%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|2|7|28.57%|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|56|96|58.33%|
|:large_blue_circle:|com/starrocks/sql/ast/CreateMaterializedViewStatement.java|5|7|71.43%|
|:large_blue_circle:|com/starrocks/catalog/Database.java|15|16|93.75%|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|4|4|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 82 / 132 (62.12%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|2|7|28.57%|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|56|96|58.33%|
|:large_blue_circle:|com/starrocks/sql/ast/CreateMaterializedViewStatement.java|5|7|71.43%|
|:large_blue_circle:|com/starrocks/catalog/Database.java|15|16|93.75%|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|4|4|100.00%|

**Review by kangkaisen — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java [hunk 1]
```diff
@@ -130,6 +135,8 @@ private void genColumnAndSetIntoStmt(CreateMaterializedViewStatement statement,
             List<Expr> outputExpression = queryRelation.getOutputExpression();
             for (int i = 0; i < outputExpression.size(); ++i) {
                 Column column = new Column(columnOutputNames.get(i), outputExpression.get(i).getType());
+                // set column is duplicate key
+                column.setIsKey(true);
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java:139]: are all columns key?

File: fe/fe-core/src/main/java/com/starrocks/persist/EditLog.java [hunk 2]
```diff
@@ -201,6 +202,13 @@ public static void loadJournal(GlobalStateMgr globalStateMgr, JournalEntity jour
                     globalStateMgr.replayDropTable(db, info.getTableId(), info.isForceDrop());
                     break;
                 }
+                case OperationType.OP_CREATE_MATERIALIZED_VIEW: {
+                    CreateTableInfo info = (CreateTableInfo) journal.getData();
+                    LOG.info("Begin to unprotect create materialized view. db = "
+                            + info.getDbName() + " create materialized view = " + info.getTable().getId());
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/persist/EditLog.java:208]: better to add table name into log

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java (hunk 1 above)
Comment (blackstar-baba) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java:139]: I will limit it with short key count

**Review by ABingHuang — APPROVED** [review]
LGTM

**Review by kangkaisen — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Pull request merged**